### PR TITLE
Exams

### PR DESCRIPTION
**Pull request by knvlewel**
Please ignore and every commit with '**Merge**' or '**Revert**'

### CODE REVIEW

File: .vs/ProjectSettings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "CurrentProjectSetting": null
+}
```
Comment (tshemsedinov) [inline .vs/ProjectSettings.json:3]: Don't commit different crap

File: Tasks/types.js [hunk 2]
```diff
@@ -1,24 +1,14 @@
 // Count types in array
+'use strict'
```
Comment (tshemsedinov) [inline Tasks/types.js:2]: Semicolon?

File: Tasks/types.js [hunk 3]
```diff
@@ -1,24 +1,14 @@
...
 
-types_=function(s) {
-  types_=(
-  {
-  "number": 0,
-  "string": 0,
-  "boolean": 0,
+const typeCounter = (array) => {
+  let typesList = {};
+  for (let elem of array) {
```
Comment (tshemsedinov) [inline Tasks/types.js:6]: Prefer const

File: Tasks/types.js [hunk 4]
```diff
@@ -1,24 +1,14 @@
...
+    const currentType = typeof elem;
+    typesList[currentType] === undefined ? typesList[currentType] = 1 : typesList[currentType]++;
```
Comment (tshemsedinov) [inline Tasks/types.js:8]: Improve readability

File: Tasks/validate.js [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
+  if (typeof name === 'string') {
+    const regEx = /[A-Z]/;
+    const words = name.split(' ');
+    for (let word of words) {
+      if (regEx.test(word)) continue;
+      else return false;
+    }
+    return true;
+  }
+  else return false;
```
Comment (tshemsedinov) [inline Tasks/validate.js:14]: Use `return early` principle

File: Tasks/validate.js [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
-  for (C of T) {
-    if (C === ' ') continue;
-    if (C.toLowerCase().charCodeAt(0) >= 97 && C.toLowerCase().charCodeAt(0) <= 122) { } else { return false; }
-  };
-  return true;
-};
-};
+const validName = (name) => {
+  if (typeof name === 'string') {
+    const regEx = /[A-Z]/;
```
Comment (tshemsedinov) [inline Tasks/validate.js:6]: Rewrite without regexp

File: Tasks/validate.js [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
+    const words = name.split(' ');
+    for (let word of words) {
+      if (regEx.test(word)) continue;
+      else return false;
```
Comment (tshemsedinov) [inline Tasks/validate.js:10]: Try to swap if/else branches and you will avoid else at all

**Review by knvlewel — COMMENTED** [review]
If you see something with 'Revert' or 'Merge', ignore them. I wanted to delete these commits, but it only got worse, there were duplicates and now I get an error in IDE. If you can help remove these commits i would be very grateful

**Pull request closed without merging**